### ISSUE TITLE
Guzzle v3 causing issues with other packages requiring Guzzle v4

### ISSUE THREAD

**Issue by Harrisonbro** [php-vcr/php-vcr#78]
I'm trying to use VCR with another package which uses `"guzzlehttp/guzzle": "4.2.2"` but because VCR is using `"guzzle/http": "3.*"` I'm getting an error:

```
To set a CURLOPT_READFUNCTION, CURLOPT_INFILESIZE must be set.
```

I've spoken to the maintainer of the other package, [`omniphx/forrest`](https://github.com/omniphx/forrest), in some depth about it in [issue 17](https://github.com/omniphx/forrest/issues/17) on his repository and full details of the errors we've managed to reproduce are available [there](https://github.com/omniphx/forrest/issues/17).

Do you have any plans to upgrade to Guzzle 4 or 5, or is there a workaround you could suggest to solve [the issue we're having](https://github.com/omniphx/forrest/issues/17)?

Thanks

**Comment by adri** [comment]
Just merged the "removal of Guzzle" branch. There are integration tests now with different Guzzle versions and Soap. Please let me know if there are problems.

**Comment by matjack1** [comment]
Hello @adri since we solved the other issue (#63) we are now using PHP-VCR in our project :) I tried then to use it also for calls to the SalesForce API and I'm experiencing similar problems to the ones in this issue. I'm continuing on this issue because I think it's the same problem. If you prefer me to open another one I can do that.

This is what I'm recording now:
```

-
    request:
        method: POST
        url: 'https://test.salesforce.com/services/Soap/c/29.0/0DF20000000TuVr'
        headers:
            Host: test.salesforce.com
            Content-Type: 'text/xml; charset=utf-8; action=""'
        body: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<SOAP-ENV:Envelope xmlns:SOAP-ENV=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:ns1=\"urn:enterprise.soap.sforce.com\"><SOAP-ENV:Body><ns1:login><ns1:username>YYY</ns1:username><ns1:password>XXX</ns1:password></ns1:login></SOAP-ENV:Body></SOAP-ENV:Envelope>\n"
    response:
        status:
            http_version: '1.1'
            code: '500'
            message: 'Server Error'
        headers:
            Date: 'Tue, 03 Feb 2015 15:04:21 GMT'
            Set-Cookie: 'BrowserId=CZr-GXpiRN2d7TLherhWeQ;Path=/;Domain=.salesforce.com;Expires=Sat, 04-Apr-2015 15:04:21 GMT'
            Expires: 'Thu, 01 Jan 1970 00:00:00 GMT'
            Content-Type: text/xml;charset=UTF-8
            Content-Length: '287'
        body: '<?xml version="1.0" encoding="UTF-8"?><soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/"><soapenv:Body><soapenv:Fault><faultcode>soapenv:Client</faultcode><faultstring>SOAPAction HTTP header missing</faultstring></soapenv:Fault></soapenv:Body></soapenv:Envelope>'
```
Looks like the problem is related to some missing header. I've tried both with and without the headers matcher, but no joy. Do you have any suggestion? Thanks!

**Comment by adri** [comment]
Maybe the authentication header. There has been work on this recently, are you sure you are on the latest vcr version?
